Libs/DICOM/Widgets/ctkDICOMBrowser.cpp

```diff
@@ -828,16 +878,203 @@ void ctkDICOMBrowser::onSeriesRightClicked(const QPoint &point)
     {
     foreach (const QString& uid, selectedSeriesUIDs)
       {
       d->DICOMDatabase->removeSeries(uid);
       }
     }
+  else if (selectedAction == exportAction)
+    {
+    ctkFileDialog* directoryDialog = new ctkFileDialog();
+    directoryDialog->setOption(QFileDialog::DontUseNativeDialog);
+    directoryDialog->setOption(QFileDialog::ShowDirsOnly);
+    directoryDialog->setFileMode(QFileDialog::DirectoryOnly);
+    bool res = directoryDialog->exec();
+    if (res)
+      {
+      QStringList dirs = directoryDialog->selectedFiles();
+      QString dirPath = dirs[0];
+      this->exportSelectedSeries(dirPath, selectedSeriesUIDs);
+      }
+    delete directoryDialog;
+    }
+}
+
+//----------------------------------------------------------------------------
+void ctkDICOMBrowser::exportSelectedSeries(QString dirPath, QStringList uids)
+{
+  Q_D(ctkDICOMBrowser);
+
+  foreach (const QString& uid, uids)
+    {
+    QStringList filesForSeries = d->DICOMDatabase->filesForSeries(uid);
+
+    // Use the first file to get the overall series information
+    QString firstFilePath = filesForSeries[0];
+    QHash<QString,QString> descriptions (d->DICOMDatabase->descriptionsForFile(firstFilePath));
+    QString patientName = descriptions["PatientsName"];
+    QString patientIDTag = QString("0010,0020");
+    QString patientID = d->DICOMDatabase->fileValue(firstFilePath, patientIDTag);
+    QString studyDescription = descriptions["StudyDescription"];
+    QString seriesDescription = descriptions["SeriesDescription"];
+    QString studyDateTag = QString("0008,0020");
+    QString studyDate = d->DICOMDatabase->fileValue(firstFilePath,studyDateTag);
+    QString seriesNumberTag = QString("0020,0011");
+    QString seriesNumber = d->DICOMDatabase->fileValue(firstFilePath,seriesNumberTag);
+
+    QString sep = "/";
+    QString nameSep = "-";
+    QString destinationDir = dirPath + sep + patientID;
+    if (!patientName.isEmpty())
+      {
+      destinationDir += nameSep + patientName;
+      }
+    destinationDir += sep + studyDate;
+    if (!studyDescription.isEmpty())
+      {
+      destinationDir += nameSep + studyDescription;
+      }
+    destinationDir += sep + seriesNumber;
+    if (!seriesDescription.isEmpty())
+      {
+      destinationDir += nameSep + seriesDescription;
+      }
+    destinationDir += sep;
+
+    // make sure only ascii characters are in the directory path
+    destinationDir = destinationDir.toLatin1();
+    // replace any question marks that were used as replacements for non ascii
+    // characters with underscore
+    destinationDir.replace("?", "_");
+
+    // create the destination directory if necessary
+    if (!QDir().exists(destinationDir))
+      {
+      if (!QDir().mkpath(destinationDir))
+        {
+        QString errorString =
+          QString("Unable to create export destination directory:\n\n")
+          + destinationDir;
+        ctkMessageBox createDirectoryErrorMessageBox;
```

Comment:
Instead of displaying a `ctkMessageBox` for each failure. May be you could collect all errors into a QStringLists. The function `exportSelectedSeries` could return a `bool` equal to `false` if at least on series failed to export. 

Then, in case of failure the user could call `lastErrors()` to return the `QStringList` container all errors.

The advantage of this approach is that you would have a nice decoupling of logic and display of error in a GUI ...